ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilterInputField.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+
+import { SearchInput, SelectOption } from '@patternfly/react-core';
+import { SelectedEntity } from './EntitySelector';
+import { SelectedAttribute } from './AttributeSelector';
+import {
+    CompoundSearchFilterConfig,
+    SearchFilterAttribute,
+    SelectSearchFilterAttribute,
+} from '../types';
+
+import CheckboxSelect from './CheckboxSelect';
+
+export type InputFieldValue = string | number | undefined | string[];
+export type InputFieldOnChange = (value: InputFieldValue) => void;
+
+export type CompoundSearchFilterInputFieldProps = {
+    selectedEntity: SelectedEntity;
+    selectedAttribute: SelectedAttribute;
+    value: InputFieldValue;
+    onSearch: InputFieldOnChange;
+    onChange: InputFieldOnChange;
+    config: Partial<CompoundSearchFilterConfig>;
+};
+
+function isSelectType(
+    attributeObject: SearchFilterAttribute
+): attributeObject is SelectSearchFilterAttribute {
+    return attributeObject.inputType === 'select';
+}
+
+function CompoundSearchFilterInputField({
+    selectedEntity,
+    selectedAttribute,
+    value,
+    onSearch,
+    onChange,
+    config,
+}: CompoundSearchFilterInputFieldProps) {
+    if (!selectedEntity || !selectedAttribute) {
+        return null;
+    }
+
+    const attributeObject: SearchFilterAttribute =
+        config[selectedEntity]?.attributes[selectedAttribute];
+
+    if (!attributeObject) {
+        return null;
+    }
+
+    if (attributeObject.inputType === 'text') {
+        const textLabel = `Filter results by ${attributeObject.filterChipLabel.toLowerCase()}`;
+        return (
+            <SearchInput
+                aria-label={textLabel}
+                placeholder={textLabel}
+                value={value as string}
+                onChange={(_event, _value) => onChange(_value)}
+                onSearch={(_event, _value) => onSearch(_value)}
+                onClear={() => onChange('')}
+            />
+        );
+    }
+    if (isSelectType(attributeObject)) {
+        const attributeLabel = attributeObject.displayName.toLowerCase();
+        const selection = value as string[];
```

Comment:
Instead of using `as string[]` we should use a util that converts any `value` that is passed in to a `string[]`. Potentially just creating `[]` if the value is a `number`/`undefined`.

I think we could run into issues as-is if `value` is equal to `''`, which is the default, when we call `includes()` on it down below.

Reply:
Good idea. I'll make a function that returns only a `string[]`